src/exp.jl

```diff
@@ -1,121 +1,119 @@
+    return @horner(x, 1.0f0, 0.6931472f0, 0.24022652f0, 
+                        0.05550327f0, 0.009617995f0, 
+                        0.0013400431f0, 0.00015478022f0)
 end
-
-@inline function exp10_kernel(x::FloatType32)
-    c6 = 0.2064004987f0
-    c5 = 0.5417877436f0
-    c4 = 0.1171286821f1
-    c3 = 0.2034656048f1
-    c2 = 0.2650948763f1
-    c1 = 0.2302585125f1
-    return @horner x c1 c2 c3 c4 c5 c6
+@inline function expb_kernel(::Val{ℯ}, x::Union{Float32, SVec{W, Float32} where W})
+    return @horner(x, 1.0f0, 1.0f0, 0.5f0,
+                        0.16666415f0, 0.041666083f0,
+                        0.008375129f0, 0.0013956056f0)
+end
+@inline function expb_kernel(::Val{10}, x::Union{Float32, SVec{W, Float32} where W})
+    return @horner(x, 1.0f0, 2.3025851f0, 2.6509492f0,
+                        2.034648f0, 1.1712388f0, 
+                        0.54208815f0, 0.20799689f0)
 end
 
-"""
-    exp10(x)
-
-Compute the base-`10` exponential of `x`, that is `10ˣ`.
-"""
-@inline function exp10(d::V) where {V <: FloatType}
-    T = eltype(d)
-    q = round(T(MLOG10_2) * d)
-    qi = unsafe_trunc(fpinttype(T), q)
-
-    s = muladd(q, -L10U(T), d)
-    s = muladd(q, -L10L(T), s)
-
-    u = exp10_kernel(s)
-    u = V(dnormalize(dadd(T(1.0), dmul(u,s))))
-
-    u = ldexp2k(u, qi)
+const J_TABLE= Float64[2.0^(big(j-1)/256) for j in 1:256]
 
-    u = vifelse(d > max_exp10(T), T(Inf), u)
-    u = vifelse(d < min_exp10(T), T(0.0), u)
+@noinline function exp_fallback(::Val{2}, x::SVec{W,T}) where {W, T}
+    return SVec(ntuple(Val(W)) do i Core.VecElement(exp2(x[i])) end)
+end
+@noinline function exp_fallback(::Val{ℯ}, x::SVec{W,T}) where {W, T}
+    return SVec(ntuple(Val(W)) do i Core.VecElement(exp(x[i])) end)
+end
+@noinline function exp_fallback(::Val{10}, x::SVec{W,T}) where {W, T}
+    return SVec(ntuple(Val(W)) do i Core.VecElement(exp10(x[i])) end)
+end
 
-    return u
+for (func, base) in (:exp2=>Val(2), :exp=>Val(ℯ), :exp10=>Val(10))
+    @eval begin
+        @inline function ($func)(x::SVec{W,T}) where {W, T<:Float64}
+            any(!(abs(x) < MAX_EXP($base, T))) && return exp_fallback($base, x)
+            N_float = vmuladd(x, LogBo256INV($base, T), MAGIC_ROUND_CONST(T))
+            N = reinterpret(SVec{W,Int64}, N_float) % Int32
+            N_float = vsub(N_float, MAGIC_ROUND_CONST(T))
+            
+            r = vmuladd(N_float, LogBo256U($base, T), x)
+            r = vmuladd(N_float, LogBo256L($base, T), r)
+            js = vload(stridedpointer(J_TABLE), (N&255,))
+            k = N >> 8
+            
+            small_part = reinterpret(SVec{W, Int64}, vmuladd(js, expm1b_kernel($base, r), js))
+            twopk = rem(vadd(k,53), UInt64) << 52
+            return reinterpret(SVec{W, T}, vadd(twopk, small_part))
+        end
+        
+        @inline function ($func)(x::SVec{W,T}) where {W, T<:Float32}
+            any(!(abs(x) < MAX_EXP($base, T))) && return exp_fallback(x)
```

Comment:
I think it'd be better to replace these with `vifelse` statements at the end.
E.g.,
```julia
res = ...
res = vifelse(x > MAX_EXP($base, T), T(Inf), res)
res = vifelse(x < MIN_EXP($base, T), zero(T), res)
return res
```

Reply:
The thought on this was that given how rare over/underflow is, it would make sense to have outlined error throwing. That said, I'll time it and keep the better version.

Reply:
This would almost certainly be faster than `exp_fallback`, meaning you could perhaps define `exp_fallback` in this manner.

Reply:
The `vifelse` versions are slightly faster on my computer.
```julia
julia> @btime myexp($(Ref(sx32))[]) # any -> branch
  4.672 ns (0 allocations: 0 bytes)
SVec{16,Float32}<57738.16f0, 0.18027152f0, 331.4781f0, 19.489355f0, 36.489277f0, 2.390847f-6, 0.0051179463f0, 0.00011763656f0, 2.3629273f-6, 238.08304f0, 1.8774968f-5, 0.00017372977f0, 0.05602656f0, 1.0087299f0, 0.0019804446f0, 10.530331f0>

julia> @btime myexp($(Ref(sx32))[]) # vifelse
  4.435 ns (0 allocations: 0 bytes)
SVec{16,Float32}<57738.16f0, 0.18027152f0, 331.4781f0, 19.489355f0, 36.489277f0, 2.390847f-6, 0.0051179463f0, 0.00011763656f0, 2.3629273f-6, 238.08304f0, 1.8774968f-5, 0.00017372977f0, 0.05602656f0, 1.0087299f0, 0.0019804446f0, 10.530331f0>
```